### ISSUE TITLE
Script terminates when retrieving vCenter licensing information in some vSphere 6.0 environments

### ISSUE THREAD

**Issue by tpcarman** [AsBuiltReport/AsBuiltReport.VMware.vSphere#34]
**Describe the bug**
When generating a VMware vSphere As Built Report against a vSphere 6.0 environment, the script terminates with the following error `New-AsBuiltReport : You cannot call a method on a null-valued expression.`

The issue has been isolated to the `Get-License` function when retrieving vCenter Server license information. It also appears that the problem is isolated to some vSphere 6.0 environments.

**Expected behavior**
VMware vSphere report should be generated without error and show valid vCenter Server licensing information.

**System (please provide the following information about the system from which you are trying to generate a report):**
 - Windows PowerShell version [Provide output from the following command: `$PSVersionTable.PSVersion`] 5.1.17763.316
 - 3rd party PowerShell module name and version [e.g. VMware PowerCLI 11.2] VMware.VimAutomation.Core 11.2.0.12483638
 - AsBuiltReport Core module version [e.g. 1.0.0] 1.0.0
 - AsBuiltReport Report module name and version [e.g. AsBuiltReport.VMware.vSphere 1.0.0] 1.0.0

**Additional context**

**Comment by tpcarman** [comment]
Issue has been identified with resolution in the code below currently in testing.

```function Get-License {
        <#
    .SYNOPSIS
    Function to retrieve vSphere product licensing information.
    .DESCRIPTION
    Function to retrieve vSphere product licensing information.
    .NOTES
    Version:        0.1.2
    Author:         Tim Carman
    Twitter:        @tpcarman
    Github:         tpcarman
    .PARAMETER VMHost
    A vSphere ESXi Host object
    .PARAMETER vCenter
    A vSphere vCenter Server object
    .PARAMETER Licenses
    All vSphere product licenses
    .INPUTS
    System.Management.Automation.PSObject.
    .OUTPUTS
    System.Management.Automation.PSObject.
    .EXAMPLE
    PS> Get-License -VMHost ESXi01
    .EXAMPLE
    PS> Get-License -vCenter VCSA
    .EXAMPLE
    PS> Get-License -Licenses
    #>
        [CmdletBinding()][OutputType('System.Management.Automation.PSObject')]

        Param
        (
            [Parameter(Mandatory = $false, ValueFromPipeline = $false)]
            [ValidateNotNullOrEmpty()]
            [PSObject]$vCenter, 
            [PSObject]$VMHost,
            [Parameter(Mandatory = $false, ValueFromPipeline = $false)]
            [Switch]$Licenses
        ) 

        $LicenseObject = @()
        $ServiceInstance = Get-View ServiceInstance -Server $vCenter
        $LicenseManager = Get-View $ServiceInstance.Content.LicenseManager -Server $vCenter
        $LicenseManagerAssign = Get-View $LicenseManager.LicenseAssignmentManager -Server $vCenter
        if ($VMHost) {
            $VMHostId = $VMHost.Extensiondata.Config.Host.Value
            $VMHostAssignedLicense = $LicenseManagerAssign.QueryAssignedLicenses($VMHostId)    
            $VMHostLicense = $VMHostAssignedLicense.AssignedLicense
            if ($VMHostLicense.LicenseKey -and $Options.ShowLicenseKeys) {
                $VMHostLicenseKey = $VMHostLicense.LicenseKey
            } else {
                $VMHostLicenseKey = "*****-*****-*****" + $VMHostLicense.LicenseKey.Substring(17)
            }
            $LicenseObject = [PSCustomObject]@{                               
                Product = $VMHostLicense.Name 
                LicenseKey = $VMHostLicenseKey                   
            }
        }
        if ($vCenter) {
            #$vCenterAssignedLicense = $LicenseManagerAssign.QueryAssignedLicenses($vCenter.InstanceUuid)
            $vCenterAssignedLicense = $LicenseManagerAssign.GetType().GetMethod("QueryAssignedLicenses").Invoke($LicenseManagerAssign,@($_.MoRef.Value)) | Where-Object {$_.EntityID -eq $vCenter.InstanceUuid}
            $vCenterLicense = $vCenterAssignedLicense.AssignedLicense
            if ($vCenterLicense.LicenseKey -and $Options.ShowLicenseKeys) { 
                $vCenterLicenseKey = $vCenterLicense.LicenseKey
            } else {
                $vCenterLicenseKey = "*****-*****-*****" + $vCenterLicense.LicenseKey.Substring(17)
            }
            $LicenseObject = [PSCustomObject]@{                               
                Product = $vCenterLicense.Name
                LicenseKey = $vCenterLicenseKey
            }
        }
        if ($Licenses) {
            foreach ($License in $LicenseManager.Licenses) {
                if ($Options.ShowLicenseKeys) {
                    $LicenseKey = $License.LicenseKey
                } else {
                    $LicenseKey = "*****-*****-*****" + $License.LicenseKey.Substring(17)
                }
                $Object = [PSCustomObject]@{                               
                    'Product' = $License.Name
                    'LicenseKey' = $LicenseKey
                    'Total' = $License.Total
                    'Used' = $License.Used                     
                }
                $LicenseObject += $Object
            }
        }
        Write-Output $LicenseObject
    }```

**Comment by tpcarman** [comment]
Resolved in #40
